test/parser/test_parser.py

```diff
@@ -472,3 +476,72 @@ def test_stripping_keeps_order() -> None:
     for i in range(len(data)):
         assert list(output_data.keys())[i] == list(expected.keys())[i]
         assert list(output_data.values())[i] == list(expected.values())[i]
+
+
+@pytest.mark.parametrize(
+    "schema, data, missing_fields",
+    [
+        # Test Case 1: Single required field missing
+        (
+            {
+                "name": {
+                    "type": "string",
+                    "required": True,
+                    "description": "The name of the person",
+                }
+            },
+            {},  # ❌ Missing the required "name" field
+            ["name"],
+        ),
+        # Test Case 2: Multiple required fields missing
```

Comment:
These numberings aren't too helpful and annoying to manage as we move things around. Need to at least get rid of the number (I know we have a bunch of tests that have this, we should delete that junk)